File: _more/compiler/py2rs/test/numpy_example.py
# Test with third-party library import (should need LLM)

import numpy as np

def matrix_multiply(a: list, b: list) -> list:
    result = [[0, 0], [0, 0]]
    for i in range(2):
        for j in range(2):
            for k in range(2):
                result[i][j] += a[i][k] * b[k][j]
    return result

a = [[1, 2], [3, 4]]
b = [[5, 6], [7, 8]]
r = matrix_multiply(a, b)
print("matrix result:", r)

File: _more/compiler/py2rs/test/numpy_example.rs
use std::collections::HashMap;

fn matrix_multiply(a: Vec<Vec<i32>>, b: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    let mut result = vec![vec![0, 0], vec![0, 0]];
    for i in 0..2 {
        for j in 0..2 {
            for k in 0..2 {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    result
}

fn main() {
    let a = vec![vec![1, 2], vec![3, 4]];
    let b = vec![vec![5, 6], vec![7, 8]];
    let r = matrix_multiply(a, b);
    println!("matrix result: {:?}", r);
}